### PR TITLE
feat: Support adding documentRoot vfile context to MDC parser

### PR DESCRIPTION
**Pull request by JessicaSachs**
### 🔗 Linked issue

#274 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### Problem
Some remark plugins rely on the standardized vfile context being attached to the file object. Currently, the MDC parser doesn't support passing this context through, which limits compatibility with these plugins. See #274 

### Solution
- Added `rootDocument` option to `inlineOptions` in the markdown parser creation
- This allows passing through vfile context without breaking existing functionality
- Chose to add at the `inlineOptions` level rather than within `parseMarkdown` to minimize changes to function signatures

### Implementation Details
- Extended `inlineOptions` interface to include `rootDocument` option
- Modified the processing stream to merge `rootDocument` with the existing context:
  ```typescript
  const mdcWithContext = { ...(inlineOptions.rootDocument || {}), value: content, data: frontmatter }
  ```
- Companion change needed in `@nuxt/content` to resolve and pass `documentRoot` (already working locally)

### Impact
- Non-breaking change
- Improves compatibility with remark plugins that depend on vfile context
- Maintains existing parser behavior while adding new capability

### Related Changes
- A follow-up PR will be opened in `@nuxt/content` to complete the implementation

### CODE REVIEW

**Comment by JessicaSachs** [comment]
The VFile API is comfortingly universal and browser-safe, so I'm not nervous about using it.

**Comment by JessicaSachs** [comment]
No idea why `typecheck` is failing and I can't get the `@nuxtjs/mdc` types to resolve locally when trying to run `pnpm typecheck` on my computer. Plz halp.

File: src/runtime/parser/index.ts [hunk 1]
```diff
@@ -97,7 +97,8 @@ export const createMarkdownParser = async (inlineOptions: MDCParseOptions = {})
     const { content, data: frontmatter } = await parseFrontMatter(md)
 
     // Start processing stream
-    const processedFile = await processor.process({ value: content, data: frontmatter })
+    const mdcWithContext = { ...(inlineOptions.rootDocument || {}), value: content, data: frontmatter }
+    const processedFile = await processor.process(mdcWithContext)
```
Comment (farnabaz) [inline src/runtime/parser/index.ts:101]: VFile is directly related to the current processing task. Passing a predefined `rootDocument` for all processing tasks may cause invalid results. Especially with Nuxt Content's multi-source feature.

Instead, we should define a second argument for  `return async (md: string): Promise<MDCParserResult>` function to receive `VFile` data.

```ts
  return async (md: string, vFileData: VFile = {}): Promise<MDCParserResult> => {
    // Extract front matter data
    const { content, data: frontmatter } = await parseFrontMatter(md)

    // Start processing stream
    const processedFile = await processor.process({ ...vFileData, value: content, data: frontmatter })
    ...
  }
```

**Review by farnabaz — APPROVED** [review]
👍

**Pull request merged**